hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -666,6 +659,11 @@ private void finalizeOrRollbackMigration(PartitionRuntimeState partitionState) {
         }
 
         updateAllPartitions(partitionTable);
+        partitionStateManager.setVersion(partitionState.getVersion());
+        if (!partitionStateManager.setInitialized()) {
```

Comment:
Ok. I was actually wondering if at this point where we are applying new partition state the partition state manager could still be uninitialized or is the partition state manager initialized from earlier in the node lifecycle.